src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacRegexStringMatchValidator.java

```diff
@@ -0,0 +1,131 @@
+            log.error("Question does not have any answers. " + question.getId() + " src: "
+                    + question.getCanonicalSourceFile());
+
+            feedback = new Content("This question does not have any correct answers");
+        }
+
+        // STEP 1: Did they provide an answer at all?
+
+        if (null == feedback && (null == userAnswer.getValue() || userAnswer.getValue().isEmpty())) {
+            feedback = new Content("You did not provide an answer");
+        }
+
+        // STEP 2: If they did, does their answer match a known answer?
+
+        if (null == feedback) {
+
+            // Sort the choices so that we match incorrect choices last, taking precedence over correct ones.
+            List<Choice> orderedChoices = getOrderedChoices(regexStringMatchQuestion.getChoices());
+
+            // For all the choices on this question...
+            for (Choice c : orderedChoices) {
+
+                // ... that are of the RegexPattern type, ...
+                if (!(c instanceof RegexPattern)) {
+                    log.error("Isaac RegexStringMatch Validator for questionId: " + regexStringMatchQuestion.getId()
+                            + " expected there to be a RegexPattern. Instead it found a Choice.");
+                    continue;
+                }
+                RegexPattern regexPattern = (RegexPattern) c;
+
+                if (null == regexPattern.getValue() || regexPattern.getValue().isEmpty()) {
+                    log.error("Expected a regex pattern to match on, but none found in choice for question id: "
+                            + regexStringMatchQuestion.getId());
+                    continue;
+                }
+
+                // ... check if they match the pattern, ...
+                if (matchesPattern(regexPattern.getValue(), userAnswer.getValue(), regexPattern.isCaseInsensitive(),
+                        regexPattern.isMultiLineRegex())) {
+                    // ... and break at the first matched pattern.
+                    feedback = (Content) regexPattern.getExplanation();
+                    responseCorrect = regexPattern.isCorrect();
+                    break;
+                }
+            }
+        }
+
+        return new QuestionValidationResponse(question.getId(), userAnswer, responseCorrect, feedback, new Date());
+    }
+
+    private boolean matchesPattern(String trustedRegexPattern, String userValue, final Boolean caseInsensitive,
+                                   final Boolean multiLineRegex) {
+
+        if (null == trustedRegexPattern || null == userValue) {
+            return false;
+        }
+
+        // The pattern is case sensitive and single line by default
+        Pattern answerPattern = Pattern.compile(trustedRegexPattern,
+                (null != caseInsensitive && caseInsensitive ? Pattern.CASE_INSENSITIVE : 0)
+                        | (null != multiLineRegex && multiLineRegex ? Pattern.MULTILINE : 0));
```

Comment:
This bitwise OR hiding in between the flags is actually the way to combine multiple regex flags and not part of the logic of the inline conditionals. Our old CheckStyle was dead against inline conditionals, but I think these are probably fine.
Adding to the comment above that "the flags are combined with a bitwise or" could be helpful though.